Tasks/ServiceFabricDeployV1/ServiceFabricSDK/Utilities.ps1

```diff
@@ -298,7 +298,7 @@ function Wait-ServiceFabricApplicationUpgradeAction
         return $upgradeStatus;
     }
 
-    $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted") }
+    $upgradeStatusValidator = { param($upgradeStatus) return !($upgradeStatus.UpgradeState -eq "RollingBackCompleted" -or $upgradeStatus.UpgradeState -eq "RollingForwardCompleted" -or $upgradeStatus.UpgradeState -eq "Failed" -or $upgradeStatus.UpgradeState -eq "Invalid") }
```

Comment:
This change make sense. Curious to know if we've seen any case where UpgraeState came out as FAILED, which eventually means that Service Fabric Cluster failed to invoke failure action given that they have retries in place for the same.